R/deploy.R

```diff
@@ -487,49 +487,65 @@ has_image <- function(content) {
 #'
 #' Set the Content Image using a variety of methods.
 #'
-#' NOTE: `set_image_webshot()` requires [webshot2::webshot()], but currently
-#' skips and warns for any content that requires authentication until the
-#' `webshot2` package supports authentication.
-#'
 #' @param content A content object
-#' @param path The path to an image on disk
-#' @param url The url for an image
-#' @param ... Additional arguments passed on to [webshot2::webshot()]
+#' @param path A file path or URL to an image
 #'
-#' @rdname set_image
 #' @family content functions
+#' @rdname set_content_image
 #' @export
-set_image_path <- function(content, path) {
-  warn_experimental("set_image_path")
+set_content_image <- function(content, path) {
+  warn_experimental("set_content_image")
   validate_R6_class(content, "Content")
-  guid <- content$get_content()$guid
 
+  valid_path <- NULL
+  if (file.exists(path)) {
+    valid_path <- path
+  } else {
+    parsed <- httr::parse_url(path)
+    if (parsed$scheme %in% c("http", "https")) {
+      valid_path <- fs::file_temp(pattern = "image", ext = fs::path_ext(parsed[["path"]]))
```

Comment:
We might use `fs` elsewhere, but I wonder if we need to use it at all here as opposed to `tempfile()`? Do we need to maintain the extension? It's a tempfile anyway so for local filesystem reasons we don't, but it's possible it's required on connect?

Reply:
Yeah, `fs` was in here before, but we don't need it for temporary files.